src/server/controllers/removeAccountController.js

```diff
@@ -20,12 +21,21 @@ const intervalForRemovalOfOldUnverifiedAccountsMillis = 1000 * 60 * 60 * 24 * 1;
  * @param {object} req - The request object.
  * @param {object} res - The response object.
  */
-function removeAccount(req, res) {
+async function removeAccount(req, res) {
     const usernameLowercase = req.params.member.toLowerCase();
+    const usernameCaseSensitive = getUsernameCaseSensitive(usernameLowercase); // False if the member doesn't exist
 
     // Check to make sure they're logged in
     if (req.user !== usernameLowercase) return res.status(403).json({'message' : 'Forbidden. This is not your account.'});
 
+		// Check that the password is correct
+		const hashedPassword = getHashedPassword(usernameLowercase);
+		const match = await bcrypt.compare(req.body.password, hashedPassword);
```

Comment:
Use the new `testPasswordForRequest()` method I added in authController instead of using repeat logic here.

Reply:
Hmm it may be redundant. You can set req.body.username on the server side, that's fine!